symphony-bdk-spring/symphony-bdk-app-spring-boot-starter/src/main/java/com/symphony/bdk/app/spring/controller/AppAuthController.java

```
@@ -0,0 +1,112 @@
+
+  @Value("${jwt-cookie.enable:false}")
+  private Boolean jwtCookieEnable;
+
+  private final ExtensionAppAuthenticator extensionAppAuthenticator;
+  private final SymphonyBdkCoreProperties properties;
+
+  public AppAuthController(SymphonyBdkCoreProperties properties, ExtensionAppAuthenticator extensionAppAuthenticator) {
+    this.properties = properties;
+    this.extensionAppAuthenticator = extensionAppAuthenticator;
+  }
+
+  @PostMapping("authenticate")
+  public ResponseEntity<AppAuthSession> authenticate(@RequestBody AppInfo appInfo) {
+    log.debug("App auth step 1: Initializing extension app authentication");
+
+    if (appInfo.getAppId() == null) {
+      log.debug("Rejecting authenticate request: null app ID received");
+      return ResponseEntity.badRequest().build();
+    }
+
+    if (appInfo.getAppToken() == null) {
+      log.debug("Rejecting authenticate request: null app token received");
+      return ResponseEntity.badRequest().build();
+    }
+
+    if (!appInfo.getAppId().equals(properties.getApp().getAppId())) {
+      log.debug("Rejecting authenticate request: app ID does not match");
+      return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+    }
+
+    try {
+      AppAuthSession authSession = extensionAppAuthenticator.authenticateExtensionApp(appInfo.getAppToken());
+      return ResponseEntity.ok(authSession);
+    } catch (AuthUnauthorizedException authUnauthorizedException) {
+      log.info("Authentication process failed");
+      return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+    }
+  }
+
+  @PostMapping("tokens/validate")
+  public ResponseEntity<?> validateTokens(@RequestBody AppToken appToken) {
+    log.debug("App auth step 2: Validating tokens");
+    if (extensionAppAuthenticator.validateTokens(appToken.getAppToken(), appToken.getSymphonyToken())) {
+      return ResponseEntity.ok().build();
+    }
+
+    return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
```

We should provide a `@ControllerAdvice` and throw exceptions, instead of returning `ResponseEntity`